src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,256 @@
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(null);
+        user2.setAge(null);
+        user3.setAge(null);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(null);
+        user2.setLogin(null);
+        user3.setLogin(null);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_FIRST);
+        user2.setLogin(GOOD_LOGIN_SECOND);
+        user3.setLogin(GOOD_LOGIN_THIRD);
+        user1.setPassword(null);
+        user2.setPassword(null);
+        user3.setPassword(null);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user1));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user2));
+        assertThrows(RegistrationException.class, () -> registrationService.register(user3));
+    }
+
+    @Test
+    void register_ageAboveMin_ok() throws RegistrationException {
+        User user1 = new User();
+        User user2 = new User();
+        User user3 = new User();
+        user1.setLogin(GOOD_LOGIN_SECOND);
+        user2.setLogin(GOOD_LOGIN_THIRD);
+        user3.setLogin(GOOD_LOGIN_FORTH);
+        user1.setPassword(GOOD_PASSWORD_FIRST);
+        user2.setPassword(GOOD_PASSWORD_SECOND);
+        user3.setPassword(GOOD_PASSWORD_THIRD);
+        user1.setAge(AGE_OF_EIGHTEEN);
+        user2.setAge(AGE_OF_TWENTY);
+        user3.setAge(AGE_OF_FORTY);
+        registrationService.register(user1);
+        registrationService.register(user2);
+        registrationService.register(user3);
+        assertEquals(AGE_OF_EIGHTEEN, user1.getAge());
+        assertEquals(AGE_OF_TWENTY, user2.getAge());
+        assertEquals(AGE_OF_FORTY, user3.getAge());
+    }
```

Comment:
The test method `register_ageAboveMin_ok` should not rely on the `user.getAge()` to confirm successful registration. Instead, it should check that the user is actually added to the `Storage.people` collection as a result of the registration process.